upup/pkg/fi/resources.go

```diff
@@ -248,3 +248,22 @@ func WrapResource(r Resource) *ResourceHolder {
 		Resource: r,
 	}
 }
+
+type TaskDependentResource struct {
+	Resource Resource
+	Task     Task
+}
+
+var _ Resource = &TaskDependentResource{}
+var _ HasDependencies = &TaskDependentResource{}
+
+func (r *TaskDependentResource) Open() (io.Reader, error) {
+	if r.Resource == nil {
+		return nil, fmt.Errorf("resource opened before it is ready")
```

Comment:
Nit: we might want to put the task name in here to help us debug if we do hit it - but I'm guessing we don't expect to hit this.

Reply:
We don't expect to hit this: it would require a bug where the consumer of the `TaskDependentResource` fails to declare it as a dependency. I'm thinking of turning this into a panic() as the retrying behavior of the `Task` scheduler would mask any such bug.

Reply:
:+1: either way works.  `klog.Fatalf` == panic but makes it easier to add more information!